bundles/org.eclipse.emfcloud.modelserver.emf/src/org/eclipse/emfcloud/modelserver/emf/common/codecs/DICodecsManager.java

```diff
@@ -90,7 +93,7 @@ public JsonNode encode(final String modelUri, final Context context, final EObje
    @Override
    public JsonNode encode(final String modelUri, final WsContext context, final EObject eObject)
       throws EncodingException {
-      return findCodec(modelUri, context.queryParamMap()).encode(eObject);
+      return findCodec(modelUri, context).encode(eObject);
```

Comment:
Why does it seem that we handle the format argument differently for `encode`/`decode`?
As far as I can see, the two `findCodec` methods are basically doing the exactly same thing, but one is a little more nested than the other. We always expect a format parameter, and if this is not given we return either the `preferredFormat` or `ModelServerPathParametersV1.FORMAT_JSON` which result in the same String.

Can we optimize that and only go with the public method `findCodec(final String modelUri, final WsContext context)` or do you see any problems with that?

Reply:
That makes sense, yes, thanks. I already made a similar change elsewhere to align the finding of the codec in the WS session case.

(I don't know why there isn't a parallel API for the REST request context case, but that's out of scope, I'm afraid)